### PR TITLE
CATROID-1102 Refactor ProjectLoadTask to Kotlin

### PR DESCRIPTION
**Pull request by wolpra98**
Refactored the Java-class ProjectLoadTask to Kotlin. Replaced
AsyncTask with Coroutines. Added Unit Test for this class. Cleaned
up all references to old class.

This pull request adds the refactored Kotlin-class ProjectLoader, instead of the Java-class ProjectLoadTask and a Unit Test for this class.
https://jira.catrob.at/browse/CATROID-1102

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/io/asynctask/ProjectLoader.kt [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    interface ProjectLoadListener {
+        fun onLoadFinished(success: Boolean)
+    }
+}
+
+fun loadProject(projectDir: File?, context: Context?): Boolean {
+    return try {
+        ProjectManager.getInstance().loadProject(projectDir, context)
+        val project = ProjectManager.getInstance().currentProject
+        DeviceVariableAccessor(projectDir).cleanUpDeletedUserData(project)
```
Comment (reichli) [inline catroid/src/main/java/org/catrobat/catroid/io/asynctask/ProjectLoader.kt:85]: I commented out the calls to these three components to see if there are tests that check whether or not the clean up happens during the loading of a project. At least the in Unit Tests and Instrumented Unit Tests there weren't any failing tests.
I think this should be covered.

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/SceneListFragment.java [hunk 2]
```diff
@@ -54,7 +54,7 @@
 import static org.catrobat.catroid.common.Constants.Z_INDEX_BACKGROUND;
 import static org.catrobat.catroid.common.SharedPreferenceKeys.SHOW_DETAILS_SCENES_PREFERENCE_KEY;
 
-public class SceneListFragment extends RecyclerViewFragment<Scene> implements ProjectLoadTask.ProjectLoadListener {
+public class SceneListFragment extends RecyclerViewFragment<Scene> implements ProjectLoader.ProjectLoadListener {
```
Comment (reichli) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/SceneListFragment.java:57]: I saw that the ProjectLoader was removed from this class as part of a different ticket. Is there any need for the ProjectLoadListener here now?

**Review by reichli — CHANGES_REQUESTED** [review]
Hi :)
I found a few things/questions and added them in the "Files changed" pane.

**Review by reichli — CHANGES_REQUESTED** [review]
It looks very good - thank you very much!

Only one thing remains: The static code analysis of the Jenkins run found some problems.
Mostly whitespace format errors and unused imports. Can you please clean those up?

**Comment by wolpra98** [comment]
Fixed the static code analysis errors.

**Pull request merged**